build/unix/compiledata.sh

```diff
@@ -73,29 +73,37 @@ CXXFLAGS=`echo $CXXFLAGS | sed -e 's/-Werror //g' -e 's/-Werror=\S* //g' -e 's/-
 BXX="`basename $CXX`"
 COMPILERVERS="$BXX"
 case $BXX in
-g++*)
+g++* | c++*)
    cxxTemp=`$CXX -dumpversion`
+   COMPILERVERSSTR=`$CXX --version 2>&1 | grep -i gcc`
```

Comment:
Couldn't c++ be clang++ (it is on MacOS)?

Reply:
right ... handled 2 lines below ... maybe ... the question is now "Can a Linux Clang build have an executable named c++?"

Reply:
Yes, theoretically it can but I've not seen it yet. I would propose to try this and see where it breaks and then fix it. Currently it is also broken as c++ is not handled and the version return by compiledata.h is wrong.

Reply:
Fair enough.  I guess it is fine for the PR but we should really auto-detect which actual compiler c++ is ....